lib/dialects/postgres/connection-manager.js

```diff
@@ -221,7 +222,8 @@ class ConnectionManager extends AbstractConnectionManager {
         dataTypes.postgres.GEOMETRY,
         dataTypes.postgres.HSTORE,
         dataTypes.postgres.GEOGRAPHY,
-        dataTypes.postgres.ENUM
+        dataTypes.postgres.ENUM,
+        dataTypes.CITEXT
```

Comment:
`dataTypes.postgres.CITEXT` is not available, right?

As we need to dynamically update OIDs it is better to define this type in `postgres/data-types.js`

```js
  function CITEXT(options) {
    if (!(this instanceof CITEXT)) return new CITEXT(options);
    BaseTypes.CITEXT.apply(this, arguments);
  }
  inherits(CITEXT, BaseTypes.CITEXT);

  BaseTypes.CITEXT.types.postgres = {
    oids: [],
    array_oids: []
  };
```

and have it exported. So you can use `dataTypes.postgres.CITEXT` here and in code just below

Reply:
Perhaps it is bit much to redeclare in the Postgres data-types? If you are concerned about how it looks, this would be cleaner:

```js
      // Reset OID mapping for dynamic type
      [
        dataTypes.GEOMETRY,
        dataTypes.HSTORE,
        dataTypes.GEOGRAPHY,
        dataTypes.ENUM,
        dataTypes.CITEXT
      ].forEach(type => {
        type.types.postgres.oids = [];
        type.types.postgres.array_oids = [];
      });

      for (const row of result.rows) {
        let type;

        if (row.typname === 'geometry') {
          type = dataTypes.GEOMETRY;
        } else if (row.typname === 'hstore') {
          type = dataTypes.HSTORE;
        } else if (row.typname === 'geography') {
          type = dataTypes.GEOGRAPHY;
        } else if (row.typname === 'citext') {
          type = dataTypes.CITEXT;
        } else if (row.typtype === 'e') {
          type = dataTypes.ENUM;
        }
```